src/trans/gpu/internal/fsc_mod.F90

```diff
@@ -116,9 +120,9 @@ SUBROUTINE FSC(ALLOCATOR,HFSC,PREEL_COMPLEX, KF_FS, KF_UV, KF_SCALARS, KUV_OFFSE
 !*       1.1      U AND V.
 #ifdef OMPGPU
 !$OMP TARGET TEAMS DISTRIBUTE PARALLEL DO COLLAPSE(3) DEFAULT(NONE) &
-!$OMP& PRIVATE(IGLG,IOFF_LAT,IOFF_UV,ZACHTE2) &
-!$OMP& SHARED(IBEG,IEND,IINC,KF_UV,R,OFFSET_VAR,G,D,KF_FS,KUV_OFFSET,F,PREEL_COMPLEX) &
-!$OMP& MAP(TO:IBEG,IEND,IINC,KF_UV,OFFSET_VAR,KF_FS,KUV_OFFSET,ZACHTE2)
+!$OMP& PRIVATE(IGLG,IOFF_LAT,IOFF_UV,ZACHTE2,JM,JF,KGL) &
+!$OMP& FIRSTPRIVATE(IBEG,IEND,IINC,OFFSET_VAR,KF_UV,KUV_OFFSET,KF_FS) &
```

Comment:
What's the reason to have these variables as `FIRSTPRIVATE` instead of `SHARED`? I ask because none of the variables in my `SHARED` clause are modified in the loop below, therefore is it not safe to have them shared rather than have every thread make its own copy?

Reply:
I wanted to keep OpenACC and OpenMP as similar as possible with regards to overall behavior, but yes `SHARED` here would be safe as well. I'm reviewing this module more carefully after realizing this comment:  https://github.com/ecmwf-ifs/ectrans/pull/206#discussion_r1939638430